sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+symbols using the BL instruction. A symbol is not pre-emptable if:
+
+ * It has ``STB_LOCAL`` binding.
+
+ * It has a symbol visibility that is not ``STV_DEFAULT``.
+
+ * It has been given the ``local`` symbol version, defined in
+   SYM-VER_.
+
+All function calls to pre-emptable symbols must be indirected through
+the static linker created Procedure Linkage Table.
+
+Procedure Linkage Table
+-----------------------
+
+Much as the global offset table (GOT) redirects position independent
+address calculations to absolute locations, the Procedure Linkage
+Table (PLT) redirects position independent function calls to absolute
+locations. The static linker cannot resolve execution transfers (such
+as function calls) from one executable or shared object to
+another. Consequently, the static linker arranges to have the program
+transfer control to entries in the PLT. In the AArch64 architecture,
+PLTs reside in the shared text segment, but they in turn use addresses
+in the per-process global offset table section ``.got.plt``. The
+dynamic linker determines the destinations' absolute addresses and
+modifies the global offset table entry accordingly. The dynamic linker
+can thus redirect the entries without compromising the position
+independence and shareability of the program's text. Executable files
+and shared objects have their own separate procedure linkage
+tables. The same procedure linkage table format is used for both
+executables and shared objects.
+
+Following the steps below, the dynamic linker and the program
+cooperate to resolve symbolic references to functions through the
+procedure linkage table and the global offset table.
+
+ * When first creating the memory image of the program, the dynamic
+   linker sets the second and the third entries in the global offset
+   table to special values. The steps below explain more about these
+   values.
+
+ * Each shared object file in the process image has its own PLT, and
+   control transfers to a PLT entry only from within the same ELF
+   module.
+
+ * For illustration, assume the program calls ``name1``, which transfers
+   control to the Nth PLT entry in ``PLT[N]``.
+
+ * The ``PLT[N]`` entry loads the address of the GOT entry for
+   ``name1`` (``.got.plt[N]``) into register ``ip0``, and then loads
+   and jumps to the address held in that table entry using a ``BR``
+   instruction. Initially the GOT entry holds the address of the
+   ``PLT[0] entry``, the first entry in the PLT, not the real address
+   of ``name1``.
+
+ * Now the ``PLT[0]`` entry pushes the ``ip0`` and ``lr`` registers
+   onto the stack.  The dynamic linker will later use the stacked
+   value of ``ip0`` to compute a relocation index by subtracting the
+   base of the ``GOT`` The index will be used to select an entry from
+   the table of relocation entries addressed by the ``DT_JMPREL``
+   dynamic section entry. The designated relocation entry must have
```

Comment:
I'll need to check, the original text used ``GOT`` without making the distinction between `.got` and `.got.plt` so I could have missed this one. I'm pretty sure you're right and it will be ``PLTGOT``.